functions/src/common/datastore.ts

```diff
@@ -132,25 +132,57 @@ export class Datastore {
     return this.db_.doc(job(surveyId, jobId)).get();
   }
 
-  fetchSubmissionsByJobId(surveyId: string, jobId: string) {
-    return this.db_
-      .collection(submissions(surveyId))
-      .where(sb.jobId, '==', jobId)
-      .get();
+  fetchAccessibleSubmissionsByJobId(
+    surveyId: string,
+    jobId: string,
+    userId?: string
+  ) {
+    if (!userId) {
+      return this.db_
+        .collection(submissions(surveyId))
+        .where(sb.jobId, '==', jobId)
+        .get();
+    } else {
+      return this.db_
+        .collection(submissions(surveyId))
+        .where(sb.jobId, '==', jobId)
+        .where(sb.ownerId, '==', userId)
+        .get();
+    }
   }
 
   fetchLocationOfInterest(surveyId: string, loiId: string) {
     return this.fetchDoc_(loi(surveyId, loiId));
   }
 
-  fetchLocationsOfInterestByJobId(
+  async fetchAccessibleLocationsOfInterestByJobId(
     surveyId: string,
-    jobId: string
-  ): Promise<QuerySnapshot<DocumentData, DocumentData>> {
-    return this.db_
-      .collection(lois(surveyId))
-      .where(l.jobId, '==', jobId)
-      .get();
+    jobId: string,
+    userId?: string
+  ): Promise<DocumentData[]> {
+    if (!userId) {
+      return (
+        await this.db_
+          .collection(lois(surveyId))
+          .where(l.jobId, '==', jobId)
+          .get()
+      ).docs;
```

Comment:
Is this function being called elsewhere in the application currently? I am assuming `.docs` being added at the end here might cause some issues.

Reply:
nope, it is called only there. something similar have been done by data-store.service.ts under web, but there I could use combineLatest to merge results. I didn't find a better way. what do you think @gino-m?

```
  getAccessibleLois$(
    {id: surveyId}: Survey,
    userId: string,
    canManageSurvey: boolean
  ): Observable<List<LocationOfInterest>> {
    if (canManageSurvey) {
      return this.db
        .collection(`${SURVEYS_COLLECTION_NAME}/${surveyId}/lois`)
        .valueChanges({idField: 'id'})
        .pipe(map(this.toLocationsOfInterest));
    }

    const importedLois = this.db.collection(
      `${SURVEYS_COLLECTION_NAME}/${surveyId}/lois`,
      ref => ref.where(l.source, '==', Source.IMPORTED)
    );

    const fieldDataLois = this.db.collection(
      `${SURVEYS_COLLECTION_NAME}/${surveyId}/lois`,
      ref =>
        ref
          .where(l.source, '==', Source.FIELD_DATA)
          .where(l.ownerId, '==', userId)
    );

    return combineLatest([
      importedLois.valueChanges({idField: 'id'}),
      fieldDataLois.valueChanges({idField: 'id'}),
    ]).pipe(
      map(([predefinedLois, fieldDataLois]) =>
        this.toLocationsOfInterest(predefinedLois.concat(fieldDataLois))
      )
    );
  }
```

Reply:
Lgtm

Reply:
Is the idea here that you're trying run both queries in parallel? Is the additional complexity of all the stream logic worth it? Perhaps this is an over-optimization?